core/src/test/scala/com/salesforce/op/stages/impl/tuning/DataSplitterTest.scala

```diff
@@ -56,6 +57,32 @@ class DataSplitterTest extends FlatSpec with TestSparkContext with SplitterSumma
     train.count() shouldBe dataCount
   }
 
+  it should "down-sample when the data count is above the max allowed" in {
+    val numRows = MaxTrainingSampleDefault * 2
+    val data =
+      RandomRDDs.normalVectorRDD(sc, numRows, 3, seed = seed)
+        .map(v => (1.0, Vectors.dense(v.toArray), "A")).toDF()
+    dataSplitter.preValidationPrepare(data)
+
+    val dataBalanced = dataSplitter.validationPrepare(data)
+    // validationPrepare calls the data sample method that samples the data to a target ratio but there is an epsilon
+    // to how precise this function is which is why we need to check around that epsilon
+    val samplingErrorEpsilon = (0.1 * MaxTrainingSampleDefault).toLong
+
+    dataBalanced.count() shouldBe MaxTrainingSampleDefault +- samplingErrorEpsilon
+  }
+
+  it should "set and get maxTrainingSample" in {
+    val numRows = 2000
+    val maxRows = numRows / 2
+
+    dataSplitter
+      .setReserveTestFraction(0.0)
+      .setMaxTrainingSample(maxRows)
+
+    dataSplitter.getMaxTrainingSample shouldBe maxRows
+  }
+
```

Comment:
It's probably worth checking `downSampleFraction` params was set here as well, for completeness so that you have checked everything in `DataSplitterParams`

Reply:
made changes here 433d4836b19bdc74596708e3f5de5061cfc7d242